### PR TITLE
RTCP Implementation

### PR DESCRIPTION
**Pull request by moninom1**
Issue #, if available:

RTCP implementation.

Description of changes:

Add RTCP message serializer/deserializer along with message parser for different type of packets received. The change also adds TWCC packet manager.

Testing -

- RTCP tests run with the wrapper layer for serialization and deserialization and feedback packet parsing.
- Unit test added for TWCC manager to validate the functionality.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice

### CODE REVIEW

**Review by aggarg — DISMISSED** [review]

File: source/include/rtcp_api.h [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                                         const RtcpSenderReport_t * pSenderReport,
+                                         uint8_t * pBuffer,
+                                         size_t * pBufferLength );
+
+RtcpResult_t Rtcp_SerializeReceiverReport( RtcpContext_t * pCtx,
+                                           const RtcpReceiverReport_t * pReceiverReport,
+                                           uint8_t * pBuffer,
+                                           size_t * pBufferLength );
+
+RtcpResult_t Rtcp_DeSerializePacket( RtcpContext_t * pCtx,
```
Comment (ActoryOu) [inline source/include/rtcp_api.h]:
```suggestion
RtcpResult_t Rtcp_DeserializePacket( RtcpContext_t * pCtx,
```

File: source/include/rtcp_data_types.h [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ <--+
+ */
+typedef struct RtcpRembPacket
+{
+    uint32_t senderSsrc;
+    uint32_t mediaSourceSsrc;
+    uint32_t bitRateMantissa;
+    uint8_t bitRateExponent;
+    uint32_t * pSsrcList;
+    size_t ssrcListLength;
```
Comment (hassanctech) [inline source/include/rtcp_data_types.h]: We can make `ssrcListLength` a `uint8_t` since it is 8 bits.

File: source/include/rtcp_data_types.h [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+    uint32_t lastSR;
+    uint32_t delaySinceLastSR;
+} RtcpReceptionReport_t;
+
+typedef struct RtcpSenderReport
+{
+    uint32_t senderSsrc;
+    RtcpSenderInfo_t senderInfo;
+    RtcpReceptionReport_t * pReceptionReports;
+    size_t numReceptionReports;
```
Comment (hassanctech) [inline source/include/rtcp_data_types.h]: It seems `numReceptionReports` can also be `uint8_t`.  Suppose MTU is 1500 bytes (upper bound), the reception report is 24 bytes itself. With that a crude upper bound for how many reception reports there can be is ~ 62 (1500/24).  Certainly a `uint8_t` should be sufficient for this.

File: source/include/rtcp_data_types.h [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+ *   2    :                               ...                             :
+ *        +=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+
+ *        |                  profile-specific extensions                  |
+ *        +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ */
+typedef struct RtcpReceiverReport
+{
+    uint32_t senderSsrc;
+    RtcpReceptionReport_t * pReceptionReports;
+    size_t numReceptionReports;
```
Comment (hassanctech) [inline source/include/rtcp_data_types.h]: Same here can be `uint8_t` instead of `size_t`

File: source/include/rtcp_data_types.h (hunk 2 above)
Comment (moninom1) [reply source/include/rtcp_data_types.h]: Nice catch , let me update this.

File: source/include/rtcp_data_types.h (hunk 3 above)
Comment (moninom1) [reply source/include/rtcp_data_types.h]: Yeah i agree based on MTU size, uin8_t should suffice to hold this list.

**Review by aggarg — DISMISSED** [review]

**Review by aggarg — APPROVED** [review]

**Review by hassanctech — APPROVED** [review]
nice!

**Pull request merged**